### PR TITLE
improve `ColorRangeSelector` ux

### LINKED ISSUE #27278: Color picker not working for region map question visualization
**Describe the bug**
Color picker for 'Map' question visualisation is not updating map color when different color is selected.

**To Reproduce**
1. Go to question with region map visualisation
2. Click on 'Visualisation' button
3. Click on 'Map' option
4. Click on 'Color' color picker 
5. Select different color option
6. Nothing happens

**Expected behavior**
Map visualisation should change color to selected color picker option.

**Information about your Metabase Installation:**
- Browser: Brave v1.46.144 
- Metabase version: v0.45.0

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Closes https://github.com/metabase/metabase/issues/27278

### Description

Some users pointed out that the behavior for our color picker was unintuitive, as you had to  click on the color range preview after selecting the new color for the change to actually take affect.

This PR changes the behavior to be be user friendly. Instead of having to click twice, clicking on the new color will immediately change the selected color value to it. Additional, the popover now no longer closes after making the selection, so the user can continue to browse and try out different colors.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. New question -> People -> Count by State
2. Visualization -> Map -> settings -> click on color bar
3. Clicking on a different color should immediately change the selection to it
4. Clicking on the invert button (dual arrows icon) should immediately invert the selection. When selecting different colors the selection should remain inverted.

### Demo

https://user-images.githubusercontent.com/37751258/227244746-745eb134-3ee9-4812-ae13-cc32e61e66aa.mov

Before

https://user-images.githubusercontent.com/37751258/227245966-3c4e85bb-3919-420f-95f4-30e2c4e40244.mov

After

### Checklist

- ✅ Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/core/components/ColorRangeSelector/ColorRangePopover.tsx [hunk 1]
```diff
@@ -44,30 +44,37 @@ const ColorSelectorContent = forwardRef(function ColorRangeSelector(
...
     getDefaultColor(initialValue, colors, colorMapping),
   );
 
   const [value, setValue] = useState(() =>
-    getColorRange(color, colorMapping, getInverted(initialValue, colorMapping)),
+    getColorRange(color, colorMapping, isInverted),
   );
 
-  const handleSelect = useCallback(
+  const handleColorSelect = useCallback(
```
Comment (EmmadUsmani) [inline frontend/src/metabase/core/components/ColorRangeSelector/ColorRangePopover.tsx:59]: Updated the name of this and a couple other methods to be a bit more descriptive

File: frontend/src/metabase/core/components/ColorRangeSelector/ColorRangePopover.tsx [hunk 2]
```diff
@@ -44,30 +44,37 @@ const ColorSelectorContent = forwardRef(function ColorRangeSelector(
...
-    [colorMapping],
-  );
+      const newValue = getColorRange(newColor, colorMapping, isInverted);
 
-  const handleChange = useCallback(
-    (newValue: string[]) => {
+      setColor(newColor);
+      setValue(newValue);
       onChange?.(newValue);
-      onClose?.();
```
Comment (EmmadUsmani) [inline frontend/src/metabase/core/components/ColorRangeSelector/ColorRangePopover.tsx:66]: For a better UX we don't close the popover, it's better to leave it open so the user can seamlessly try different colors

File: frontend/src/metabase/core/components/ColorRangeSelector/ColorRangeSelector.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+
+import { color } from "metabase/lib/colors";
+import ColorRangeSelector from "./ColorRangeSelector";
+
+const DEFAULT_VALUE = [color("white"), color("brand")];
+const DEFAULT_COLORS = [color("brand"), color("summarize"), color("filter")];
+
+interface TestColorRangeSelectorProps {
```
Comment (ranquild) [inline frontend/src/metabase/core/components/ColorRangeSelector/ColorRangeSelector.unit.spec.tsx:10]: nit: this is a bit different to the usual pattern:

```
const setup = () => { 
  const onChange = jest.fn();
  render(<ColorRangeSelector ... onChange={onChange} />);

  return { onChange }; 
});

it("....", () => {
  const { onChange } = setup(); 
});
```

the pattern is also documented in the frontend docs in the repo

File: frontend/src/metabase/core/components/ColorRangeSelector/ColorRangeSelector.unit.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+describe("ColorRangeSelector", () => {
+  it("should call `onChange` upon clicking a color", async () => {
+    const onChange = jest.fn();
+    render(<TestColorRangeSelector onChange={onChange} />);
+
+    screen.getByRole("button").click();
+    expect(await screen.findByRole("tooltip")).toBeInTheDocument();
+
+    screen.getByLabelText(color("summarize")).click();
+    expect(onChange).toHaveBeenCalled();
```
Comment (ranquild) [inline frontend/src/metabase/core/components/ColorRangeSelector/ColorRangeSelector.unit.spec.tsx:33]: I think the test suite does't test that you can select multiple colors without closing the popover.

**Review by ranquild — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Thanks @EmmadUsmani !

**Pull request merged**